### ISSUE TITLE
"Find" window in Node Graph is not floating (i3wm)

### ISSUE THREAD

**Issue by z3ntu** [MrKepzie/Natron#1316]
When I press Ctrl+F (or click "Find" in the Node Graph), the popup window is handled like a normal window (split the screen with the main window) and not like a popup/floating window. I also can't manually add a rule in my i3 config, as the window disappears when you lose focus (so I can't run xprop to find out a name or something about the window).

**Comment by olear** [comment]
Natron has been tested to work in "normal" window managers. Running 'xprop' on the find window is no problem (run xprop, alt+tab to Natron, ctrl+f, click on new window), so you should be able to add whatever rules you need.

**Comment by z3ntu** [comment]
Actually as soon as I get the focus to another window, than either the main window of Natron or the popup search window, the search/find window closes.

**Comment by olear** [comment]
You should define Natron as floating.

```
for_window [class="Natron-bin" instance="Natron-bin"] floating enable
```

**Comment by z3ntu** [comment]
Doesn't work... And this would probably set all Natron windows to floating, which is also not the wanted behaviour.

**Comment by olear** [comment]
As I said earlier Natron has not been designed to work in i3 or similar window managers, and "find" is not the only window affected. If you want Natron to work as intended you need to add an exception in your i3 config. We might consider doing some changes that will make Natron integrate better at a later date, but currently this has low priority.

**Comment by z3ntu** [comment]
The line which causes the non-floating of the window is (probably) https://github.com/MrKepzie/Natron/blob/master/Gui/NodeGraph45.cpp#L294 .
I will try to find (hopefully all) non-floating - should-be-floating windows by testing out the application and create a pull request afterwards.

**Comment by devernay** [comment]
So should it be `Qt::Dialog | Qt::CustomizeWindowHint`  instead?
The example to test Qt flags is in the Qt sources, located at `examples/widgets/windowflags`

**Comment by devernay** [comment]
Or is `Qt::Tool | Qt::CustomizeWindowHint` more appropriate?

**Comment by z3ntu** [comment]
I think popup is more appropriate.

**Comment by devernay** [comment]
just did a PR : https://github.com/MrKepzie/Natron/pull/1318
@z3ntu tell me what you think of it

**Comment by devernay** [comment]
Natron is still using Qt4, but the code has to be Qt5-compatible as much as possible.

**Comment by olear** [comment]
Fixed in #1318
